crates/sui-core/src/authority/authority_store_pruner.rs

```diff
@@ -746,6 +773,71 @@ impl AuthorityStorePruner {
     }
 }
 
+#[derive(Clone)]
+pub struct ObjectsCompactionFilter {
+    db: Weak<AuthorityPrunerTables>,
+    metrics: Arc<ObjectCompactionMetrics>,
+}
+
+impl ObjectsCompactionFilter {
+    pub fn new(db: Arc<AuthorityPrunerTables>, registry: &Registry) -> Self {
+        Self {
+            db: Arc::downgrade(&db),
+            metrics: ObjectCompactionMetrics::new(registry),
+        }
+    }
+    pub fn filter(&mut self, key: &[u8], value: &[u8]) -> anyhow::Result<Decision> {
+        let ObjectKey(object_id, version) = bcs::from_bytes(key)?;
+        let object: StoreObjectWrapper = bcs::from_bytes(value)?;
```

Comment:
@sadhansood: just double checking if we still need this deserialization. 
it's used to verify that object is not Deleted/Wrapped, but watermark is populated from [`effects.modified_at_versions()`](https://github.com/MystenLabs/sui/blob/abde857436dac8e02529e9a05a76d0c4fae92295/crates/sui-core/src/authority/authority_store_pruner.rs#L143), so they shouldn't be GCd even without this check. Is this correct?

Reply:
Yeah I think they shouldn't be GC'd but perhaps safe to keep it unless we are absolutely sure about it.